core/trino-main/src/main/java/io/trino/cost/ComparisonStatsCalculator.java

```diff
@@ -239,4 +252,134 @@ private static PlanNodeStatsEstimate estimateExpressionNotEqualToExpression(
         rightExpressionSymbol.ifPresent(symbol -> result.addSymbolStatistics(symbol, rightNullsFiltered));
         return result.build();
     }
+
+    private static PlanNodeStatsEstimate estimateExpressionToExpressionInequality(
+            ComparisonExpression.Operator operator,
+            PlanNodeStatsEstimate inputStatistics,
+            SymbolStatsEstimate leftExpressionStatistics,
+            Optional<Symbol> leftExpressionSymbol,
+            SymbolStatsEstimate rightExpressionStatistics,
+            Optional<Symbol> rightExpressionSymbol,
+            boolean isNonEstimatablePredicateApproximationEnabled)
+    {
+        if (leftExpressionStatistics.isUnknown() || rightExpressionStatistics.isUnknown()) {
+            return PlanNodeStatsEstimate.unknown();
+        }
+
+        StatisticRange leftRange = StatisticRange.from(leftExpressionStatistics);
+        StatisticRange rightRange = StatisticRange.from(rightExpressionStatistics);
+        if (leftRange.isEmpty() || rightRange.isEmpty()) {
+            return inputStatistics.mapOutputRowCount(rowCount -> 0.0);
+        }
+        // We don't know the correlation between NULLs, so we take the max nullsFraction from the expression statistics
+        // to make a conservative estimate for nullsFilterFactor
+        double nullsFilterFactor;
+        if (isNaN(leftExpressionStatistics.getNullsFraction()) && isNaN(rightExpressionStatistics.getNullsFraction())) {
+            nullsFilterFactor = 1.0;
```

Comment:
nit: I think we might just bail in this case, but I'm fine leaving it too

Reply:
updated to bail in this case